### ISSUE TITLE
A11y: Set `alt=""` on course badges and some other illustrative images

### ISSUE THREAD

**Issue by Asartea** [TheOdinProject/theodinproject#3993]
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `brief description of request` format, e.g. `Add dark mode to website`

The following checkbox is OPTIONAL:

-   [x] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**
Currently all of the course badges have an alt text of `course name badge`, but this does sounds weird when read by a screenreader and also serves no real purpose, as this info is always also conveyed by an accompanying header. These images instead should have their alt text set to none (`alt=""`) to prevent screenreaders from reading them.

There are also several other images for which this is an issue:
- Succes story avatars
- How it works images
- What you can expect at The Odin Project images on `/about`
- Why you should get involved images on `/contributing`
- probably a few more that I have missed

**2. Acceptance Criteria:**

```[tasklist]
### Tasks
- [ ] All decorative images have an alt of `""`
- [ ] All none-decorative images have a clearly descriptive alt
```

**Comment by luuu-xu** [comment]
Hi @Asartea I would like to work on this a11y image alt issue! I will focus on image alt only and scan through our pages and make changes.

**Comment by thatblindgeye** [comment]
@Asartea For the course badges, you're talking about how, for example, in the following screenshot:

![Intermediate HTML and CSS course badge and header](https://github.com/TheOdinProject/theodinproject/assets/70952936/8f77981c-33cb-4b7f-a73b-13b0ddae2f5a)

The badge has an `img` element with alt text of "Intermediate HTML and CSS badge"?

If so, then we don't want to make the alt text an empty string. Since the badge is inside a link, we need the image to have an alt text both to provide an accessible name for the link, and to display visible text should the image not load. What we *should* do is remove the "badge" text from the alt text, though. So "Intermediate HTML and CSS" instead of "Intermediate HTML and CSS badge".

Whether we really need these badges to also be links is another question. I would also think that we should convey the progress displayed in a badge as well (which again is another matter).

For the other spots you mentioned, yeah we should update the alt text for those to be an empty string.

**Comment by thatblindgeye** [comment]
@luuu-xu it looks like @Asartea has asked to be assigned the issue as part of creating the issue. Let's see if they would need to split the work up at all, otherwise they'll be assigned to work on it.

**Comment by Asartea** [comment]
@thatblindgeye feel free to assign it to them; always happy to see someone else working on a11y stuff

**Comment by thatblindgeye** [comment]
@luuu-xu you're assigned!

**Comment by luuu-xu** [comment]
@thatblindgeye Hi, I wanted to get an opinion from you.

> Since the badge is inside a link, we need the image to have an alt text both to provide an accessible name for the link, and to display visible text should the image not load. What we should do is remove the "badge" text from the alt text, though. So "Intermediate HTML and CSS" instead of "Intermediate HTML and CSS badge".

I understand and agree on this change and have made these changes sitewide. But when I am checking the course badges on our index page, where there are 9 courses in a flexbox under "Learn everything you need to know".

The screenreader will read these links to be "link, heading level 3, ruby, ruby" because now the badge `img` has `alt="ruby"` and the `h3` has a textcontent of "ruby" too. Should I change these badge images' alt to be empty `alt=""` ? Because I feel like if the images are not loaded, the information is presented by the h3 so now the alt is kind of redundant.

**Comment by thatblindgeye** [comment]
@luuu-xu excellent catch! I'd agree that for those particular badges `alt=""` makes sense. Awesome flex of those a11y muscles 💪🏼

**Comment by luuu-xu** [comment]
@thatblindgeye Thank you! I think I've got everything and I am creating the PR now. I am still new to improving a11y so I would love to learn and work with you on more of these issues too!

**Comment by luuu-xu** [comment]
@thatblindgeye @Asartea Hi, I am not sure who I should ask for a review for the PR I submitted. Let me know! Thank you!

**Comment by Asartea** [comment]
I'm not a maintainer, and so I can't give a review as GitHub means it, but I will note that you seem to have some linter errors you'll need to fix before it gets merged

**Comment by ManonLef** [comment]
I have requested a review on the PR from a maintainer
